lib/src/disposable/disposable.dart

```diff
@@ -273,9 +280,13 @@ class Disposable implements _Disposable, DisposableManagerV3 {
     _internalDisposables.add(disposable);
 
     completer.future.catchError((e) {
-      _internalDisposables.remove(disposable);
+      if (!isDisposedOrDisposing) {
+        _internalDisposables.remove(disposable);
+      }
     }).then((_) {
-      _internalDisposables.remove(disposable);
+      if (!isDisposedOrDisposing) {
+        _internalDisposables.remove(disposable);
+      }
```

Comment:
🔮  You could use `whenComplete` to handle the error and success case in one statement.